diff --git a/CMakeLists.txt b/CMakeLists.txt
index 489bec1..424cd4d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,6 +17,10 @@ add_subdirectory(lv_drivers)
 
 SET(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/bin)
 
+file(GLOB_RECURSE UI_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/../example/main/ui/*.c)
+list(APPEND SOURCES ${UI_SOURCES})
+
 add_executable(main main.c mouse_cursor_icon.c ${SOURCES} ${INCLUDES})
+target_include_directories(main SYSTEM PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/../example/main)
 target_link_libraries(main PRIVATE lvgl lvgl::examples lvgl::demos lvgl::drivers ${SDL2_LIBRARIES})
 add_custom_target(run COMMAND ${EXECUTABLE_OUTPUT_PATH}/main)
diff --git a/lv_conf.h b/lv_conf.h
index 0b9a6dc..8a5c280 100644
--- a/lv_conf.h
+++ b/lv_conf.h
@@ -24,7 +24,7 @@
  *====================*/
 
 /*Color depth: 1 (1 byte per pixel), 8 (RGB332), 16 (RGB565), 32 (ARGB8888)*/
-#define LV_COLOR_DEPTH 32
+#define LV_COLOR_DEPTH 16
 
 /*Swap the 2 bytes of RGB565 color. Useful if the display has an 8-bit interface (e.g. SPI)*/
 #define LV_COLOR_16_SWAP 0
diff --git a/lv_drv_conf.h b/lv_drv_conf.h
index 4f6a4e2..ddb47f7 100644
--- a/lv_drv_conf.h
+++ b/lv_drv_conf.h
@@ -95,8 +95,8 @@
 #endif
 
 #if USE_SDL || USE_SDL_GPU
-#  define SDL_HOR_RES     480
-#  define SDL_VER_RES     320
+#  define SDL_HOR_RES     240
+#  define SDL_VER_RES     240
 
 /* Scale window by this factor (useful when simulating small screens) */
 #  define SDL_ZOOM        1
diff --git a/main.c b/main.c
index 02d6c1e..de70323 100644
--- a/main.c
+++ b/main.c
@@ -61,7 +61,7 @@ static void hal_init(void);
 /**********************
  *   GLOBAL FUNCTIONS
  **********************/
-
+#include "ui/ui.h"
 int main(int argc, char **argv)
 {
   (void)argc; /*Unused*/
@@ -91,7 +91,7 @@ int main(int argc, char **argv)
 //  lv_example_flex_3();
 //  lv_example_label_1();
 
-    lv_demo_widgets();
+    ui_clock_init();
 
   while(1) {
       /* Periodically call the lv_task handler.
